auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -417,6 +425,15 @@ public SignInIntentBuilder setIsSmartLockEnabled(boolean enabled) {
             return this;
         }
 
+        /**
+         * Links the current user to a new one.
+         */
+        public SignInIntentBuilder linkWithCurrentUser(OnMergeFailedListener onMergeFailedListener) {
```

Comment:
I find the pattern of adding a listener to an `Intent` builder strange. The result of the action in the `Intent` should be returned via `onActivityResult` if the `Activity` that fired it did so for result.